drivers/serial/uart_stm32.c

```diff
@@ -1795,11 +1868,10 @@ static int uart_stm32_async_rx_enable(const struct device *dev,
 	/* Enable RX DMA requests */
 	uart_stm32_dma_rx_enable(dev);
 
-	/* Enable IRQ IDLE to define the end of a
-	 * RX DMA transaction.
-	 */
-	LL_USART_ClearFlag_IDLE(usart);
-	LL_USART_EnableIT_IDLE(usart);
+	/* Set timeout if applicable */
+	if (timeout != SYS_FOREVER_US) {
```

Comment:
This is a behavior change vs application configuration vs initial implementation and would probably require a dedicated commit.